### PR TITLE
feat: support typst query

### PR DESCRIPTION
**Pull request by OrangeX4**
Support typst query like `typst query example.typ "<note>" --field value --one` [document](https://typst.app/docs/reference/introspection/query/#command-line-queries)

```py
values = json.loads(typst.query("hello.typ", "<note>", field="value", one=True))
```

This is useful for getting metadata from typst files.

### CODE REVIEW

File: python/typst/__init__.pyi [hunk 1]
```diff
@@ -35,6 +36,23 @@ class Compiler:
             Optional[bytes]: Return the compiled file as `bytes` if output is `None`.
         """
 
+    def query(
+        self,
+        selector: str,
+        field: Optional[str] = None,
+        one: bool = False,
+        format: Optional[str] = None,
```
Comment (messense) [inline python/typst/__init__.pyi:44]: We can use a `typing.Literal["json", "yaml"]` instead of `str` for `format`.

File: src/lib.rs [hunk 2]
```diff
@@ -2,8 +2,9 @@ use std::path::PathBuf;
 
 use pyo3::exceptions::{PyIOError, PyRuntimeError};
```
Comment (messense) [inline src/lib.rs:3]:
```suggestion
use pyo3::exceptions::{PyIOError, PyRuntimeError, PyValueError};
```

File: src/lib.rs [hunk 3]
```diff
@@ -55,6 +57,34 @@ impl Compiler {
...
+        selector: &str,
+        field: Option<&str>,
+        one: bool,
+        format: Option<&str>,
+    ) -> PyResult<String> {
+        let format = format.unwrap_or("json");
+        let format = match format {
+            "json" => SerializationFormat::Json,
+            "yaml" => SerializationFormat::Yaml,
+            _ => return Err(PyRuntimeError::new_err("unsupported serialization format")),
```
Comment (messense) [inline src/lib.rs:72]:
```suggestion
            _ => return Err(PyValueError::new_err("unsupported serialization format")),
```

**Comment by messense** [comment]
LGTM, please resolve merge conflicts, thanks!

**Comment by OrangeX4** [comment]
done :-)

**Review by messense — APPROVED** [review]

**Pull request merged**